xray/test/test_variable.py

```diff
@@ -110,6 +110,8 @@ def test_index_0d_timedelta64(self):
     def test_index_0d_not_a_time(self):
         d = np.datetime64('NaT')
         x = self.cls(['x'], [d])
+        # Wasn't able to figure out why this was failing.
+        # AssertionError: numpy.datetime64('NaT') != numpy.datetime64('NaT')
         self.assertIndexedLikeNDArray(x, d, None)
```

Comment:
The problem here is in `assertIndexedLikeNDArray` above. We need to change `assertEqual` to something that works on NaN/NaT, e.g., `assert x == y or (x != x and y != y)`.